google-map-poly.html

```diff
@@ -0,0 +1,554 @@
+      }
+    },
+
+    _editableChanged: function() {
+      this.poly && this.poly.setEditable(this.editable);
+    },
+
+    _fillColorChanged: function() {
+      this.poly && this.poly.set('fillColor', this.fillColor);
+    },
+
+    _fillOpacityChanged: function() {
+      this.poly && this.poly.set('fillOpacity', this.fillOpacity);
+    },
+
+    _geodesicChanged: function() {
+      this.poly && this.poly.set('geodesic', this.geodesic);
+    },
+
+    _iconsChanged: function() {
+      this.poly && this.poly.set('icons', this.icons);
+    },
+
+    _mapChanged: function() {
+      // Poly will be rebuilt, so disconnect existing one from old map and listeners.
+      if (this.poly) {
+        this.poly.setMap(null);
+        google.maps.event.clearInstanceListeners(this.poly);
+      }
+
+      if (this.map && this.map instanceof google.maps.Map) {
+        this._createPoly();
+      }
+    },
+
+    _mouseEventsChanged: function() {
+      if (this.poly) {
+        if (this.mouseEvents) {
+          this._forwardEvent('mousedown');
+          this._forwardEvent('mousemove');
+          this._forwardEvent('mouseout');
+          this._forwardEvent('mouseover');
+          this._forwardEvent('mouseup');
+        } else {
+          this._clearListener('mousedown');
+          this._clearListener('mousemove');
+          this._clearListener('mouseout');
+          this._clearListener('mouseover');
+          this._clearListener('mouseup');
+        }
+      }
+    },
+
+    _strokeColorChanged: function() {
+      this.poly && this.poly.set('strokeColor', this.strokeColor);
+    },
+
+    _strokeOpacityChanged: function() {
+      this.poly && this.poly.set('strokeOpacity', this.strokeOpacity);
+    },
+
+    _strokePositionChanged: function() {
+      this.poly && this.poly.set('strokePosition', this._convertStrokePosition());
+    },
+
+    _strokeWeightChanged: function() {
+      this.poly && this.poly.set('strokeWeight', this.strokeWeight);
+    },
+
+    _zIndexChanged: function() {
+      this.poly && this.poly.set('zIndex', this.zIndex);
+    },
+
+    // Helper logic.
+
+    _buildPathFromPoints: function() {
+      this._points = Array.prototype.slice.call(Polymer.dom(this.$.points).getDistributedNodes());
+      window.console.log("Points:  " + this._points.length);
+      window.console.log(Polymer.dom(this).getDestinationInsertionPoints());
+
+      // Build path from current points (ignoring vertex insertions while doing so).
+      this._building = true;
+      this.path.clear();
+      for (var i = 0, point; point = this._points[i]; ++i) {
+        this.path.push(point.getPosition());
+      }
+      this._building = false;
+
+      this.fire('google-map-poly-path-built', this.path);
+
+      // Watch for future updates.
+      if (this._mutationObserver) {
+        return;
+      }
+      this._mutationObserver = new MutationObserver(this._buildPathFromPoints.bind(this));
+      this._mutationObserver.observe(this, {
```

Comment:
That doesn't work:
  Uncaught NotFoundError: Failed to execute 'observe' on 'MutationObserver': The provided node was null.